xds/src/test/java/io/grpc/xds/XdsClientImplTest.java

```diff
@@ -0,0 +1,1373 @@
+    pattern = "foo-*";
+    assertThat(XdsClientImpl.matchHostName("bar-.googleapis.com", pattern)).isFalse();
+    assertThat(XdsClientImpl.matchHostName("foo.googleapis.com", pattern)).isFalse();
+    assertThat(XdsClientImpl.matchHostName("foo.googleapis.com", pattern)).isFalse();
+    assertThat(XdsClientImpl.matchHostName("foo-", pattern)).isFalse();
+    assertThat(XdsClientImpl.matchHostName("foo-bar.com", pattern)).isTrue();
+    assertThat(XdsClientImpl.matchHostName("foo-.com", pattern)).isTrue();
+    assertThat(XdsClientImpl.matchHostName("foo-bar", pattern)).isTrue();
+  }
+
+  private static DiscoveryResponse buildDiscoveryResponse(String versionInfo,
+      List<com.google.protobuf.Any> resources, String typeUrl, String nonce) {
+    return
+        DiscoveryResponse.newBuilder()
+            .setVersionInfo(versionInfo)
+            .setTypeUrl(typeUrl)
+            .addAllResources(resources)
+            .setNonce(nonce)
+            .build();
+  }
+
+  private static DiscoveryRequest buildDiscoveryRequest(String versionInfo,
+      String resourceName, String typeUrl, String nonce) {
+    return
+        DiscoveryRequest.newBuilder()
+            .setVersionInfo(versionInfo)
+            .setNode(NODE)
+            .setTypeUrl(typeUrl)
+            .addResourceNames(resourceName)
+            .setResponseNonce(nonce)
+            .build();
+  }
+
+  private static Listener buildListener(String name, com.google.protobuf.Any apiListener) {
+    return
+        Listener.newBuilder()
+            .setName(name)
+            .setAddress(Address.getDefaultInstance())
+            .addFilterChains(FilterChain.getDefaultInstance())
+            .setApiListener(ApiListener.newBuilder().setApiListener(apiListener))
+            .build();
+  }
+
+  private static RouteConfiguration buildRouteConfiguration(String name,
+      List<VirtualHost> virtualHosts) {
+    return
+        RouteConfiguration.newBuilder()
+            .setName(name)
+            .addAllVirtualHosts(virtualHosts)
+            .build();
+  }
+
+  private static VirtualHost buildVirtualHost(List<String> domains, String clusterName) {
+    return
+        VirtualHost.newBuilder()
+            .setName("virtualhost00.googleapis.com")  // don't care
+            .addAllDomains(domains)
+            .addRoutes(Route.newBuilder()
+                .setRoute(RouteAction.newBuilder().setCluster("whatever cluster")))
+            .addRoutes(
+                // Only the last (default) route matters.
+                Route.newBuilder()
+                    .setRoute(RouteAction.newBuilder().setCluster(clusterName)))
+            .build();
+  }
+
+  /**
+   * Matcher for DiscoveryRequest used to verify NACK requests. Eliminates the comparison of
+   * error_details for DiscoveryRequests if they are expected to be an NACK request.
+   */
+  private static class DiscoveryRequestMatcher implements ArgumentMatcher<DiscoveryRequest> {
```

Comment:
`DiscoveryRequestMatcher` -> `NackRequestMatcher`

Reply:
Changed.

Reply:
Update: I changed the name back to `DiscoveryRequestMatcher`. Instead updated the comment. Specifying this matcher being used only for NACK is a too strong requirement. It is not wrong to use this for ACK requests as well, as we do not care `error_details` field for both  ACK and NACK.

I would consider the usage of this matcher for any cases that `error_details` field does not matter.